README.md

```diff
@@ -243,22 +245,42 @@ If using make with pip, there are corresponding make targets for running linting
+
+Our commit titles have slight variations from the general Zulip style, with each:
+* starting with one or more areas in lower case, followed by a colon and space
+* area being slash-separated modified files without extensions, or the type of the change
+* ending with a concise description starting with a capital and ending with a full-stop (period)
+* having a maximum overall length of 72 (fitting github web interface without abbreviation)
+
+Some example commit titles:
+* `file1/file2/file3: Improve behavior of something.` - a general commit updating files `file1.txt`, `file2.py` and `file3.md`
+* `refactor: file1/file2: Extract some common function.` - a pure refactor which doesn't change the functional behavior
+* `bugfix: file1: Avoid some noticeable bug.` - an ideally small commit to fix a bug
+* `tests: file1: Improve test for something.` - only improve tests for `file1`, likely in `test_file1.py`
+* `requirements: Upgrade some-dependency from 9.2 to 9.3.` - upgrade a dependency from version 9.2 to version 9.3
+
+Generally with changes to code we request you update linting and tests to pass on a per-commit basis (not just per pull request).
+If you update tests, you can add eg. `Tests updated.` in your commit text.
+
+Ideally we prefer that behavioral changes are accompanied by test improvements or additions, and an accompanying `Tests added.` or similar is then useful.
+
+To make it easier to meet these style guidelines we recommend installing GitLint - see the next section!
+
+##### GitLint
 
 If you plan to submit git commits in pull-requests (PRs), then we highly suggest installing the `gitlint` commit-message hook by running `gitlint install-hook` (or `pipenv run gitlint install-hook` with pipenv setups). While the content still depends upon your writing skills, this ensures a more consistent formatting structure between commits, including by different authors.
 
 If the hook is installed as described above, then after completing the text for a commit, it will be checked by gitlint against the style we have set up, and will offer advice if there are any issues it notices. If gitlint finds any, it will ask if you wish to commit with the message as it is (`y` for 'yes'), stop the commit process (`n` for 'no'), or edit the commit message (`e` for 'edit').
 
 Other gitlint options are available; for example it is possible to apply it to a range of commits with the `--commits` option, eg. `gitlint --commits HEAD~2..HEAD` would apply it to the last few commits.
 
-**NOTE** Not all style suggestions are identified by gitlint at this time, including:
-* If modifying code (not just tests), list modified filenames without extensions between slashes in only one area of the commit title (eg. `run/model/core/README: Some description.`)
-* If modifying tests in addition to code, just note this in the body of the commit after a blank line (eg. `Tests updated.`)
```

Comment:
We don't check for this in the GitLint, right? Should we keep it here in the README?

Reply:
Some of these are mentioned in the new section, so how about adjusting the trailing sentence of that from:
```
To make it easier to meet these style guidelines we recommend installing GitLint - see the next section!
```
to
```
To aid in satisfying some of these rules you can use `GitLint`, as described in the following section.
**However**, please check your commits manually versus these style rules, since GitLint cannot check everything - or language or grammar!

Reply:
That sounds good! :+1: